.circleci/config.yml

```diff
@@ -34,11 +34,73 @@ commands:
           paths:
             - ~/.npm
             - i18n-cache/data
+  checkout-shallow:
+    steps:
+      - run:
+          name: Checkout (Shallow)
+          command: |
+            #!/bin/sh
+            set -e
+
+            # Workaround old docker images with incorrect $HOME
+            # check https://github.com/docker/docker/issues/2968 for details
+            if [ "${HOME}" = "/" ]
+            then
+              export HOME=$(getent passwd $(id -un) | cut -d: -f6)
+            fi
+
+            SSH_CONFIG_DIR="$HOME/.ssh"
+            echo "Using SSH Config Dir '$SSH_CONFIG_DIR'"
+            git --version 
+
+            mkdir -p "$SSH_CONFIG_DIR"
+            chmod 0700 "$SSH_CONFIG_DIR"
+
+            printf "%s" 'github.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ==
+            bitbucket.org ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAubiN81eDcafrgMeLzaFPsw2kNvEcqTKl/VqLat/MaB33pZy0y3rJZtnqwR2qOOvbwKZYKiEO1O6VqNEBxKvJJelCq0dTXWT5pbO2gDXC6h6QDXCaHo6pOHGPUy+YBaGQRGuSusMEASYiWunYN0vCAI8QaXnWMXNMdFP3jHAJH0eDsoiGnLPBlBp4TNm6rYI74nMzgz3B9IikW4WVK+dc8KZJZWYjAuORU3jc1c/NPskD2ASinf8v3xnfXeukU0sJ5N6m5E8VLjObPEO+mN2t/FZTMZLiFqPWc/ALSqnMnnhwrNi2rbfg/rd/IpL8Le3pSBne8+seeFVBoGqzHM9yXw==
+            ' >> "$SSH_CONFIG_DIR/known_hosts"
+            chmod 0600 "$SSH_CONFIG_DIR/known_hosts"
+
+            export GIT_SSH_COMMAND="ssh -o UserKnownHostsFile='$SSH_CONFIG_DIR/known_hosts'"
+
+            # use git+ssh instead of https
+            git config --global url."ssh://git@github.com".insteadOf "https://github.com" || true
```

Comment:
👋 @dcalhoun , awesome speed improvement with this PR! 🏃 

I wonder, is this solution based on [this ticket here](https://github.com/circleci/circleci-docs/issues/2040#issue-298727830)? It's not clear to me why one has to go the git+ssh route instead of https, which I guess makes the script need to hold the public keys for GitHub on file. Have you tried with https perhaps?

Reply:
> I wonder, is this solution based on [this ticket here](https://github.com/circleci/circleci-docs/issues/2040#issue-298727830)? 

The solution is based upon CircleCI's own default checkout script, the details of which can be found in logs of [prior CI "Checkout code" tasks](https://app.circleci.com/pipelines/github/wordpress-mobile/gutenberg-mobile/12459/workflows/09e70294-82ab-4b5e-b432-5e387cee5bc7/jobs/67166). I then used a few other public examples (e.g [one](https://discuss.circleci.com/t/shallow-clone-for-circleci-2-0-builds/20200), [two](https://discuss.circleci.com/t/custom-checkout-command/31624), [three](https://github.com/guitarrapc/git-shallow-clone-orb/blob/master/src/commands/checkout.yml), [four](https://github.com/mavenlink/orbs/blob/master/src/git/orb.yml)) as references for modifying CircleCI's default task. 

> It's not clear to me why one has to go the git+ssh route instead of https, which I guess makes the script need to hold the public keys for GitHub on file. Have you tried with https perhaps?

Utilizing SSH is likely not a requirement, but would potentially simplify —  and I _believe_ provide more secure — authentication, if future git operations require it. I do not believe we need authentication at this time (i.e. all relevant git repositories are public), but I chose to rely upon SSH to more closely align with CircleCI's default "Checkout code" script and retain the potential for simple authentication later. 

```bash
rm -f "$SSH_CONFIG_DIR/id_rsa"
printf "%s" "$CHECKOUT_KEY" > "$SSH_CONFIG_DIR/id_rsa"
chmod 0600 "$SSH_CONFIG_DIR/id_rsa"
if (: "${CHECKOUT_KEY_PUBLIC?}") 2>/dev/null; then
  rm -f "$SSH_CONFIG_DIR/id_rsa.pub"
  printf "%s" "$CHECKOUT_KEY_PUBLIC" > "$SSH_CONFIG_DIR/id_rsa.pub"
fi

export GIT_SSH_COMMAND='ssh -i "$SSH_CONFIG_DIR/id_rsa" -o UserKnownHostsFile="$SSH_CONFIG_DIR/known_hosts"'
```

The above was removed from the original CircleCI checkout script as (1) the `CHECKOUT_KEY*` environment variables appear to be special variables only available in CircleCI's own "Checkout code" take and (2) I do not believe we need authentication for the git operations we perform at this time. In the event we needed to perform authentication, we would need to reinstate similar code, but rely upon a custom environment key we set ourselves to house the SSH key. 

With that context, would you still prefer to rely upon HTTPS?